tests/README.md

```diff
@@ -22,15 +22,21 @@ Validation tests have the same requirements as integration tests however differe
 
 ### Integration vs Validation
 
-TODO
+Integration tests - Don't require any external configuration or access to any other external services. These should also be short running as they will run on every PR within CI. Integration tests do not need a configuration file to run.
+
+Validation tests - Require access to external services, and needs to a config file to run them.
 
 ### Extensions
 
 Extensions are functions that complete common operations used by tests.  Extensions should not require much configuration or support multiple behaviors.  Extensions should be simple enough that they can be used in many tests without needing to be tested themselves.
 
 ### Clients
 
-TODO
+Clients are used to create the resources in the tests. There are three clients; the rancher client, dynamic client, and the k3d client. The rancher client is used to create resources on a rancher instance. Given a host url, and a bearer token the rancher client communicates directly with the rancher instance. The dynamic client is used as means of communication on a downstream cluster, and used for kubernete resource management on that cluster. The k3d client is used to create a minimal k3d cluster, this is to give the integration tests the ability to test against a downstream cluster.
```

Comment:
I would call out that the advantage of clients is they register the cleanups with sessions.